CI failing on OS X python 3.9

From https://github.com/tskit-dev/tskit/pull/1215/checks?check_run_id=1968865308 it seems like we are getting:

```
tests/test_cli.py:523: 
125
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
126
tests/test_cli.py:63: in capture_output
127
    func(*args, **kwargs)
128
tskit/cli.py:278: in tskit_main
129
    set_sigpipe_handler()
130
tskit/cli.py:39: in set_sigpipe_handler
131
    signal.signal(signal.SIGPIPE, signal.SIG_DFL)
132
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
133

134
signalnum = <Signals.SIGPIPE: 13>, handler = <Handlers.SIG_DFL: 0>
135

136
    @_wraps(_signal.signal)
137
    def signal(signalnum, handler):
138
>       handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
139
E       ValueError: signal only works in main thread of the main interpreter
140

```

I can't replicate this on my Mac running the latest OS 11.2.1, Python (anaconda installed) 3.9.1.

--- a/python/tests/test_cli.py
+++ b/python/tests/test_cli.py
@@ -60,7 +60,10 @@ def capture_output(func, *args, **kwargs):
     sys.stderr = buffer_class()
 
     try:
-        func(*args, **kwargs)
+        # Recent versions of MacOS seem to have issues with us calling signal
+        # during tests.
+        with mock.patch("signal.signal"):
+            func(*args, **kwargs)
         stdout_output = sys.stdout.getvalue()
         stderr_output = sys.stderr.getvalue()
     finally:

